compiler/passes/src/flattening/flatten_statement.rs

```diff
@@ -110,6 +110,96 @@ impl StatementReconstructor for Flattener<'_> {
         (Statement::dummy(Default::default()), statements)
     }
 
+    /// Rewrites a console statement into a flattened form.
+    /// Console statements at the top level only have their arguments flattened.
+    /// Console statements inside a conditional statement are flattened to such that the check is conditional on
+    /// the execution path being valid.
+    /// For example, the following snippet:
+    /// ```leo
+    /// if condition1 {
+    ///    if condition2 {
+    ///        console.assert(foo);
+    ///    }
+    /// }
+    /// ```
+    /// is flattened to:
+    /// ```leo
+    /// console.assert(!(condition1 && condition2) || foo);
+    /// ```
+    /// which is equivalent to the logical formula `(condition1 ^ condition2) ==> foo`.
```

Comment:
Maybe this was meant to be
```
(condition1 /\ condition2) ==> foo
```
since `^` is exclusive disjunction in Leo.

Reply:
Ah yes, `/\` is clearer.